content/scalability/docs/control-plane.md

```diff
@@ -0,0 +1,164 @@
+# Kubernetes Control Plane
```

Comment:
Couple of other best practice suggestion - 
1. List vs Watch - we see a lot of API server issues related to agents running on worker nodes making calls to API server using "list" vs "watch" . While a lot of times, this might not be in customer's control since the agent could be a logging agent that comes from a third party but it is still valuable for customers to be aware of . See https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes

2. kube-prox in IPVS mode - See https://quip-amazon.com/EqQsAKut4sRA/MegaXL-runbook-for-customers#NAN9CA2vi1T and https://quip-amazon.com/KNN8AXKbnzKr/Enabling-IPVS-for-kube-proxy

Reply:
We had a section about watch vs list, but it wasn't actionable for customers. It was informational but only applied to customers writing custom controllers. It would probably be great if we create a EKS developer best practice guide but I'm not sure how impactful it is for scalability recommendations.

We decided against suggesting IPVS for a general audience because it lacks support for grpc and HTTP/3 which is used by many controllers so the benefits of IPVS are limited to HTTP/2 traffic which depends on workloads and controllers.

Reply:
I am fine with leaving out IPVS recommendation.

My 2 cents regarding watch vs list - even if we don't go into the level of details, I believe there is still value for customers who write custom controllers to be aware of this (even if we don't provide all the details). Even for those that are not actionable directly by customers (say they are running a 3rd party software on their EKS cluster that makes constant list call), if they know about list vs watch downside, they can start pushing/engaging the owners of the third part software  to get it resolved.

Reply:
I'm having trouble finding any reliable, public information we could provide for watch vs list. Client-go has a watch function, but at scale controllers should be using informers which provide caching and managers to reduce API load and make eventing more reliable. All of the information I found about informers is on personal blogs. The most semi-official information I found was for [kubebuilder](https://book-v1.book.kubebuilder.io/beyond_basics/using_client_go_informers.html) but this was for their old documentation and not available in the current version.

Do you have suggestions on content we could link outside of the client-go code reference?

Reply:
Let me speak with engineering to see if I can get some additional info to you.

Reply:
I believe Ugur collected some data on this when he did his comparison between Fluentd and Fluent Bit.

Reply:
Probably don't need IPVS as much now given the improvements to kube-proxy and endpoint slices. That said, there may be certain use cases where it is warranted.

Reply:
The data that Ugur collected is in this blog, https://aws.amazon.com/blogs/containers/fluentd-considerations-and-actions-required-at-scale-in-amazon-eks/ @rothgar